### PR TITLE
feat: Add .xml to file types that can be updated by ListingManager

### LINKED ISSUE #72: Handle non-cs files
we use files such as .xml files. This currently isn't being handled, and causes bugs of .xml files not being renamed.

### PR DESCRIPTION
**Pull request by BenjaminMichaelis**
Fixes #72

### CODE REVIEW

File: ListingManager/ListingManager.cs [hunk 1]
```diff
@@ -24,24 +24,30 @@ public static IEnumerable<string> GetAllExtraListings(string pathToStartFrom)
             }
         }
 
-        private static bool TryGetListing(string listingPath, out ListingInformation? listingData)
+        private static bool TryGetListing(string listingPath, out ListingInformation? listingData, bool onlyCSFiles = false)
```
Comment (Keboo) [inline ListingManager/ListingManager.cs:27]: In general `out` parameters are typically listed last.

File: ListingManager/ListingInformation.cs [hunk 2]
```diff
@@ -1,21 +1,26 @@
 using System;
+using System.Collections.Generic;
 using System.Text.RegularExpressions;
 
 namespace ListingManager
 {
-    public class ListingInformation
+    public partial class ListingInformation
     {
+        public static List<string> approvedFileTypes = new() { ".cs", ".xml" };
```
Comment (Keboo) [inline ListingManager/ListingInformation.cs:9]: I would avoid the public field and instead make it a property.
```suggestion
        public static IReadOnlyList<string> ApprovedFileTypes { get; } = new[] { ".cs", ".xml" };
```

File: ListingManager/ListingInformation.cs [hunk 3]
```diff
@@ -28,11 +33,17 @@ public ListingInformation(string listingPath)
...
             {
                 throw new ArgumentException("Listing information not successfully able to be parsed from listing path.", nameof(listingPath));
             }
-        }
+        }
+
+        [GeneratedRegex("Listing(\\d{2}).(\\d{2})([A-Za-z]*)(\\.{1}(.*))?[.cs,.xml]$")]
+        private static partial Regex ExtractListingNameFromAnyApprovedFileTypes();
+        [GeneratedRegex("Listing(\\d{2}).(\\d{2})([A-Za-z]*)(\\.{1}(.*))?.cs$")]
+        private static partial Regex ExtractListingNameFromCSFile();
```
Comment (Keboo) [inline ListingManager/ListingInformation.cs:47]: Maybe I am missing something, but why do we need both? These look to be the exact same with just the addition of the xml extension. The real program always seems to pass false. So don't we just want the false case and update the tests?

File: ListingManager/ListingManager.cs (hunk 1 above)
Comment (BenjaminMichaelis) [reply ListingManager/ListingManager.cs:27]: Removed option alltogether

File: ListingManager/ListingInformation.cs (hunk 2 above)
Comment (BenjaminMichaelis) [reply ListingManager/ListingInformation.cs:9]: Fixed

File: ListingManager/ListingInformation.cs (hunk 3 above)
Comment (BenjaminMichaelis) [reply ListingManager/ListingInformation.cs:47]: Removed Only cs file case

**Review by Keboo — APPROVED** [review]

**Pull request merged**